modules/transport-netty4/src/main/java/org/elasticsearch/http/netty4/Netty4HttpServerTransport.java

```diff
@@ -299,16 +357,35 @@ protected static class HttpChannelHandler extends ChannelInitializer<Channel> {
 
         private final Netty4HttpServerTransport transport;
         private final HttpHandlingSettings handlingSettings;
-
-        protected HttpChannelHandler(final Netty4HttpServerTransport transport, final HttpHandlingSettings handlingSettings) {
+        private final TLSConfig tlsConfig;
+        private final BiPredicate<String, InetSocketAddress> acceptChannelPredicate;
+
+        protected HttpChannelHandler(
+            final Netty4HttpServerTransport transport,
+            final HttpHandlingSettings handlingSettings,
+            final TLSConfig tlsConfig,
+            @Nullable final BiPredicate<String, InetSocketAddress> acceptChannelPredicate
+        ) {
             this.transport = transport;
             this.handlingSettings = handlingSettings;
+            this.tlsConfig = tlsConfig;
+            this.acceptChannelPredicate = acceptChannelPredicate;
         }
 
         @Override
         protected void initChannel(Channel ch) throws Exception {
             Netty4HttpChannel nettyHttpChannel = new Netty4HttpChannel(ch);
             ch.attr(HTTP_CHANNEL_KEY).set(nettyHttpChannel);
+            if (acceptChannelPredicate != null) {
+                ch.pipeline()
+                    .addFirst(
+                        "accept_channel_handler",
+                        new AcceptChannelHandler(acceptChannelPredicate, HttpServerTransport.HTTP_PROFILE_NAME)
+                    );
+            }
+            if (tlsConfig.isTLSEnabled()) {
```

Comment:
IIUC, the order here is not the opposite of previous. I'd think we'd want the accept channel handler (ip-filter) prior to tls?